src/OpenTelemetry.Instrumentation.Http/TracerProviderBuilderExtensions.cs

```diff
@@ -60,7 +60,7 @@ public static class TracerProviderBuilderExtensions
 
             configureHttpClientInstrumentationOptions?.Invoke(httpClientOptions);
 
-            builder.AddDiagnosticSourceInstrumentation((activitySource) => new HttpClientInstrumentation(activitySource, httpClientOptions));
+            builder.AddDiagnosticSourceInstrumentation(() => new HttpClientInstrumentation(httpClientOptions));
```

Comment:
```suggestion
            builder.AddInstrumentation(() => new HttpClientInstrumentation(httpClientOptions));
```

Reply:
followed by something like builder.AddLegacyActivity("nameofactivity from httpclient")

Reply:
I have created two new methods `AddInstrumentationActivitySource` and `AddLegacyActivityOperationName` which should be used in the `TracerProviderBuilderExtensions.cs` of the instrumentation projects such as AspNetCore and HttpClient.

`AddInstrumentationActivitySource` ensures that the ActivityListener's ActivityStop callback is called as the ActivitySource is updated by the OnStartActivity of instrumentation listeners.

`AddLegacyActivityOperationName` is used to limit the processing of activities in the activity listener to only those that are required.